docs: Add example of anchor and aliases in playbook (#54079)
ansible/ansible: docs/docsite/rst/user_guide/playbooks_advanced_syntax.rst

        unsafe_key: !unsafe 'unsafe value'

> Using nested variables

Reviewer: ```suggestion
Sharing variable values with YAML anchors and aliases
```

~~~~~~~~~~~~~~~~~~~~~~

> If you want to refer to the same object from different parts of playbook, you need to use anchors and aliases.

Reviewer 2: ```suggestion
If you want to refer to the same object from different parts of a playbook, you need to use anchors and aliases.
```

Reviewer 2: This would be a good point to add a link to the YAML syntax page, if you add a description there that further describes yaml anchors and aliases.

> If you want to refer to the same object from different parts of playbook, you need to use anchors and aliases.

Reviewer: ```suggestion
If you want to share variable values across tasks, `YAML anchors and aliases <https://yaml.org/spec/1.2/spec.html#id2765878>`_ help you define, maintain, and use those values in a flexible way.
```

> Anchors are denoted by ``&`` indicator while aliases are denoted by ``*``

Reviewer 2: ```suggestion
Anchors are denoted by the ``&`` indicator while aliases are denoted by ``*``
```

> Anchors are denoted by ``&`` indicator while aliases are denoted by ``*``

Reviewer: ```suggestion
You define an anchor with ``&``, then refer to it using an alias, denoted with ``*``.
```

> In a playbook, this may look like::

Reviewer: ```suggestion
Here's an example that sets three values with an anchor, uses two of those values with an alias, and overrides the third value::
```

> If you want to refer share variable values across tasks, `YAML anchors and aliases <https://yaml.org/spec/1.2/spec.html#id2765878>`_ help you define, maintain, and use those values in a flexible way.

Reviewer 2: ```suggestion
If you want to share variable values across tasks, `YAML anchors and aliases <https://yaml.org/spec/1.2/spec.html#id2765878>`_ help you define, maintain, and use those values in a flexible way.
```

    ---
    ...
    vars:
        app1:
            jvm: &jvm_opts
                opts: '-Xms1G -Xmx2G'
                port: 1000
                path: /usr/lib/app1
        app2:
            jvm:
                <<: *jvm_opts
                path: /usr/lib/app2
    ...

> Here, 'app1' and 'app2' variables shares 'opts' and 'port' information so you can anchor that

Reviewer: ```suggestion
Here, ``app1`` and ``app2`` share the values for ``opts`` and ``port`` 
```

> information using ``&jvm_opts`` and re-use this information by indicating ``<<: *jvm_opts``.

Reviewer 2: I'm having difficulty following this, as someone not familiar with YAML's support for anchor and aliases.  Initially a * meant alias, if so, what's the <<: about here?

Also not sure how I would know that the alias only reuses 'opts' and 'port'. If I put 'port: 2000' for app2, that would mean it was only sharing 'opts'?  (that is, I can reuse part of a set of variables if I set values for the other variables directly where I'm reusing as the alias?)

> information using ``&jvm_opts`` and re-use this information by indicating ``<<: *jvm_opts``.

Reviewer 2: ```suggestion
information using ``&jvm_opts`` as the anchor and re-use this information by indicating ``<<: *jvm_opts`` as the alias to that anchor.
```

> information using ``&jvm_opts`` and re-use this information by indicating ``<<: *jvm_opts``.

Reviewer: ```suggestion
using the anchor ``&jvm_opts`` and the alias ``<<: *jvm_opts``. The value for ``path`` is set by the anchor but overridden after the alias.
```

> Another example can be::

Reviewer: ```suggestion
Anchors and aliases let you share complex sets of variable values, including nested variables::
```

    ---
    - name: Using values nested inside dictionary
      hosts: localhost
      vars:
        webapp:
            app_version: &my_version 1.0
            custom_version:
                - "ToDo_App"
                - *my_version
      tasks:
      - name: Using Anchor value
        debug:
            msg: "{{ webapp.custom_version | join('-') }}"

Here, you can anchor 'app_version' value using ``&my_version`` and re-use later as ``*my_version``.

> This way you can access nested values inside dictionaries.

Reviewer 2: Confused here. I don't see *my_version used at all here? Did you mean to use msg: "{ *my_version | join('-') }} here?
